azurerm/resource_arm_application_gateway.go

```diff
@@ -0,0 +1,1685 @@
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.StandardSmall),
+								string(network.StandardMedium),
+								string(network.StandardLarge),
+								string(network.WAFLarge),
+								string(network.WAFMedium),
+							}, true),
+						},
+
+						"tier": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.Standard),
+								string(network.WAF),
+							}, true),
+						},
+
+						"capacity": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(1, 10),
+						},
+					},
+				},
+				Set: hashApplicationGatewaySku,
+			},
+
+			"disabled_ssl_protocols": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(network.TLSv10),
+						string(network.TLSv11),
+						string(network.TLSv12),
+					}, true),
+				},
+			},
+
+			"waf_configuration": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"firewall_mode": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.Detection),
+								string(network.Prevention),
+							}, true),
+						},
+
+						"rule_set_type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc:     validation.StringInSlice([]string{"OWASP"}, true),
+						},
+
+						"rule_set_version": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc:     validation.StringInSlice([]string{"2.2.9", "3.0"}, true),
+						},
+					},
```

Comment:
+1 here - this isn't a blocker by any means; so we can run with this string for now 😄